mantis-runtime/src/main/java/io/mantisrx/runtime/executor/WorkerPublisherRemoteObservable.java

```diff
@@ -225,6 +104,52 @@ public void blockUntilServerShutdown() {
         }
     }
 
+    private <K> void startKeyValueStage(KeyValueStageConfig<?, K, T> stage, Observable<Observable<T>> toServe) {
+
+        if (runNewW2WserverGroups(jobName)) {
```

Comment:
this if seems redundent (no other clause available). Maybe convert to validation?

Reply:
+1